damnit/gui/main_window.py

```diff
@@ -499,6 +509,52 @@ def _create_menu_bar(self) -> None:
         tableMenu = menu_bar.addMenu("Table")
         tableMenu.addAction(action_columns)
 
+    def export_table(self):
+        export_path, file_type = QFileDialog.getSaveFileName(self, "Export table to file",
+                                                             str(Path.home()),
+                                                             "Excel (*.xlsx);;CSV (*.csv)")
+
+        # If the user cancelled the dialog, return
+        if len(export_path) == 0:
+            return
+
+        # Make sure the path ends with the right extension
+        export_path = Path(export_path)
+        if export_path.suffix == "":
+            extension = file_type.split()[1][2:-1]
+            export_path = export_path.with_suffix(extension)
+        else:
+            extension = export_path.suffix
+
+        # Helper function to convert 2D arrays to a string tag. Meant for
+        # applying to a DataFrame.
+        def image2str(value):
+            # NOTE: when image previews are generated by the backend the
+            # summaries will change to be 3D, and this statement should be
+            # modified.
+            if isinstance(value, np.ndarray) and value.ndim == 2:
+                return "<image>"
+            else:
+                return value
+
+        # Select columns in order of their appearance in the table (note: this
+        # drops the comment_id column).
+        columns = ["Status"] + list(self.table_view.get_column_states().keys())
+        cleaned_df = self.data[columns].copy()
+        # Format timestamps nicely
+        cleaned_df["Timestamp"] = cleaned_df["Timestamp"].map(lambda x: timestamp2str(x))
+        # Format images nicely
+        cleaned_df = cleaned_df.applymap(image2str)
+
+        if extension == ".xlsx":
+            proposal = self.db.metameta["proposal"]
+            cleaned_df.to_excel(export_path, sheet_name=f"p{proposal} DAMNIT run table")
```

Comment:
Getting on my FOSS horse :horse: : if we're doing spreadsheet export, can we support OpenDocument as well as Excel? Weirdly this seems to go through the `.to_excel()` method with an `engine` parameter to say that it's actually ODS:

https://pandas.pydata.org/docs/user_guide/io.html#opendocument-spreadsheets

This would require the [odfpy](https://pypi.org/project/odfpy/) package as well.

I haven't tested this, but it looks like the additional code should be like this:

```suggestion
        if extension == ".xlsx":
            proposal = self.db.metameta["proposal"]
            cleaned_df.to_excel(export_path, sheet_name=f"p{proposal} DAMNIT run table")
        elif extension == ".ods":
            proposal = self.db.metameta["proposal"]
            cleaned_df.to_excel(export_path, engine="odf", sheet_name=f"p{proposal} DAMNIT run table")
```

Reply:
I tried this out, but for some reason the boolean `Status` column isn't handled properly :frowning_face: 
![image](https://github.com/European-XFEL/DAMNIT/assets/5361518/bad85e04-cbb3-4d1c-a43e-b1461bd3fc34)

The odfpy library also seems kinda abandoned, so I'm leaning towards not supporting ODS.

Reply:
That's disappointing, but fair enough. :-(